static/js/controllers/inbox.js

```diff
@@ -148,6 +148,16 @@ require('moment/locales');
         $scope.items = contacts || [];
       };
 
+      $scope.removeContact = function(contact) {
```

Comment:
That "removeDeletedContacts" method is historically named, but is actually for updating Messages not Contacts. Please refactor it and "mergeUpdatedContacts" to "removeDeletedMessages" and "mergeUpdatedMessages" respectively.